ebean-api/src/main/java/io/ebean/config/dbplatform/LimitOffsetSqlLimiter.java

```diff
@@ -31,12 +31,12 @@ public SqlLimitResponse limit(SqlLimitRequest request) {
     int firstRow = request.getFirstRow();
     int maxRows = request.getMaxRows();
 
-    if (maxRows > 0 || firstRow > 0) {
+    if (maxRows >= 0) {
```

Comment:
Why do you think that maxRows 0 is valid?  Why is this not `maxRows > 0` ?

Reply:
LIMIT 0 is a proper SQL expression. Not sure about the use cases but still valid.

Reply:
Hmm. What would we expect when maxRows = 0?
- empty result (so query needs not to be executed) 
- the complete result (without limit)
Both cases are valid for me.

Reply:
LIMIT 0 means it will **_always_** return an empty set. Ebean has never supported that and I'd argue it has no utility value for ORM per say.

Changing this allowing LIMIT 0 in this PR means that this PR introduces 2 changes rather than just the one that is stated - allowing LIMIT 0 is a behavior change, I'm not keen to see that included in this change.

Reply:
After a short web research I found out LIMIT 0 is sometimes used to check the query validity against the DB server. 
It seems pretty confusing that explicit setMaxRows(0) doesn't produce LIMIT 0. The other way is to restrict the argument to [1;MAX_INTEGER]

Reply:
As I see it this PR should only have the 1 change (allowing OFFSET without LIMIT).

The change to support LIMIT 0 needs to be in a separate PR. I see support for LIMIT 0 as a debatable behavior change as opposed to a bug fix so it would need to go into a major | minor version bump release and not a bug fix release etc.

Reply:
Ok. I'll split the PRs.

Reply:
@rbygrave Please check https://github.com/ebean-orm/ebean/pull/2904 and https://github.com/ebean-orm/ebean/pull/2905 PRs